### PR TITLE
Data Registry API display name modification

### PR DESCRIPTION
**Pull request by chcmsft**
# Data Plane API - Pull Request

Change the category name of the APIs from Data Registries to Data Registry.

The following is the current table of content generated from the swagger in main branch.
image
Want to change it from Data Registries to Data Registry via modifying the operation id.

The APIs are not yet public preview. Thus, no breaking change concern.

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by chcmsft** [comment]
@tjprescott Please help to take a look. Thanks.

**Comment by tjprescott** [comment]
@chcmsft as long as the REST API stewardship folks are okay with it, so am I. Will wait for Mike or Ted's approval.

**Comment by mikekistler** [comment]
@tjprescott I think this is good to go. I added the Approved-BreakingChange label.

**Review by tjprescott — APPROVED** [review]

File: specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json
Comment (tjprescott) [inline specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: It seems this needs to be changed to `DataRegistery_Get` to fix the LintDiff error that prohibits noun phrases after the underscore. cc/ @mikekistler

**Review by tjprescott — CHANGES_REQUESTED** [review]

File: specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json
Comment (mikekistler) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: I don't think "GetDataRegistry" is a noun phrase, but it has been many years since I had any formal English training so I could be wrong. I would have said that "GetDataRegistry" was a verb phrase -- a verb with a direct object.

Comment (tjprescott) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: Well, the direct object is a noun, but the point is the Swagger LintDiff rule that is failing (and blocking me from merging) is looking for Noun_Verb only, not Noun_VerbDirectObject
https://github.com/Azure/azure-rest-api-specs/pull/22583/checks?check_run_id=11305548696

Since all other operation IDs here follow that pattern, this must be the culprit.

If we want to keep this as is that's fine but we'll need someone to override the failing check and merge.

Comment (chcmsft) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: In the same swagger, there is an operationId, DataRegistry_GetContent, which passes the check.
So there may be something wrong to the LintDiff check.

@tjprescott
Can you help to find someone to override this check and merge this PR?

Comment (chcmsft) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: @tjprescott @mikekistler 
Per travis, we need someone who has the permission to bypass the fail check and merge.
Do you know who has that permission?

Thanks

Comment (mikekistler) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: I do not have merge permissions.

Comment (tjprescott) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: @chcmsft please follow the link described in the above chat:
image

We can't just force merge this otherwise every PR that is made henceforth will require a force merge.

Comment (chcmsft) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: @tjprescott
thanks for the help. I've asked in the channel.

But I think we can still force merge this PR first. I did some tests and found that other changes to this swagger won't introduce the lintDiff fail. Thus, the concern of a force merge of following PR is resolved.

Can you help to force merge this time?

Thanks

Comment (tjprescott) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: The swagger support folks should be able to do that. As I mentioned, I do not have permissions to force merge. I still think the simplest solution is to fix the operation id.

**Review by ruowan — APPROVED** [review]

File: specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json
Comment (chcmsft) [reply specification/maps/data-plane/DataRegistry/preview/2022-12-01-preview/dataregistry.json:143]: @tjprescott 
The swagger support folk had bypassed the linkdiff fail. Please help to merge. Thank you!

**Review by tjprescott — APPROVED** [review]

**Pull request merged**
